connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java

```diff
@@ -97,7 +98,11 @@ public <T> HttpResponse<T> httpRequest(String url, String method, HttpHeaders he
     public <T> HttpResponse<T> httpRequest(String url, String method, HttpHeaders headers, Object requestBodyData,
                                                   TypeReference<T> responseFormat,
                                                   SecretKey sessionKey, String requestSignatureAlgorithm) {
-        HttpClient client = httpClient();
+        // Only try to load SSL configs if we have to (see KAFKA-14816)
+        SslContextFactory sslContextFactory = url.startsWith("https://")
```

Comment:
Is this the only way to know we need to use ssl? Seems a little strange we are checking the url has https.

Reply:
Having no background here it does seem strange that we use our connect worker ssl properties for its own internal communication for rest ssl properties, we probably would want to seperate them out to their own set of properties but thats coming from someone with minimal context.  (Example here being we use our own private ca for interworker communication but may want a publicly signed ca cert for rest)

All that being said - this is a fix for a regression so no need to go ahead and change the world

Reply:
@jolshan Although it may seem rudimentary, this is the logic we were using before we made the change that broke things (see [here](https://github.com/apache/kafka/blob/ed77bebcaf8e4a8f74dd3823c905c8fd01dadf64/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java#L86)). As far as I know, there were no problems with that logic, and before we changed that part, it remained as-was for 4-5 years.

@imcdo that's already possible; you can control the SSL properties that are used for REST separately from the ones used for communication with the underlying Kafka cluster by prefixing them with `listeners.https`; see [here](https://github.com/apache/kafka/blob/a375777d8063cda087c77e520fcfd4e035e14f6f/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java#L63), [here](https://github.com/apache/kafka/blob/a375777d8063cda087c77e520fcfd4e035e14f6f/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/util/SSLUtils.java#L67), and [here](https://github.com/apache/kafka/blob/a375777d8063cda087c77e520fcfd4e035e14f6f/clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java#L316-L339).